### PR TITLE
Add Share to Cubox Plugin

### PR DESCRIPTION
**Pull request by Redwinam**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/Redwinam/obsidian-cubox

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by joethei** [comment]
Please include a link to Cubox inside your readme.

Just a note, creative commons licenses are not made for code.

[const response = await fetch(this.apiKey, {](https://github.com/Redwinam/obsidian-cubox/blob/d2f46eb4649a807500d1b708c87a966b697afce8/main.ts#L119-L125)
Use the `requestUrl` API from the Obsidian API instead, this will handle a few issues related to Obsidian automatically.

[id: "share-to-cubox",](https://github.com/Redwinam/obsidian-cubox/blob/d2f46eb4649a807500d1b708c87a966b697afce8/main.ts#L157)
You can shorten this command id, Obsidian will automatically prefix it with your plugin id, so it will be saved as "share-to-cubox:share-to-cubox".


[const apiUrl = "https://cubox.pro/c/api/save/" + this.settings.apiKey;](https://github.com/Redwinam/obsidian-cubox/blob/d2f46eb4649a807500d1b708c87a966b697afce8/main.ts#L192)
Will this URL also work when using Cubox International, or only for Cubox China?

**Comment by Redwinam** [comment]
Thank you! @joethei 

- I've now replaced `fetch` with the Obsidian API's `requestUrl` method as suggested, 
- and shortened the command id to "share this",
- and according to the Cubox International Docs (https://help.cubox.cc/save/api/), the URL `const apiUrl = "https://cubox.pro/c/api/save/" + this.settings.apiKey` will also work for Cubox International, and I have encapsulated it into the `CuboxApi` now.
- I've also included links to Cubox in my README.

Regarding the Creative Commons license, I understand it's not traditionally used for code, but I've chosen it as I prefer a non-commercial usage license. But really, I don't mind it too much 😁

And noticed there are conflicts in my Pull Request, should I open a new one to resolve these issues?

**Comment by marcusolsson** [comment]
["description": "Share Obsidian notes to Cubox using the provided API.",](https://github.com/Redwinam/obsidian-cubox/blob/128003dabff5ad6a636890190aa53a217ebd0351/manifest.json#L5)
I think you can shorten this to just "Share Obsidian notes to Cubox".

[const language = (navigator.language || "en").toLowerCase();](https://github.com/Redwinam/obsidian-cubox/blob/128003dabff5ad6a636890190aa53a217ebd0351/main.ts#L149)
This uses the locale on the user's computer, rather than the language set in Obsidian. You can use `moment.locale()` to get the Obsidian interface language.

[containerEl.createEl("h2", { text: this.plugin.translation.setting });](https://github.com/Redwinam/obsidian-cubox/blob/128003dabff5ad6a636890190aa53a217ebd0351/main.ts#L231)
https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.

**Comment by Redwinam** [comment]
@marcusolsson Thank you for your review!!

- I've updated my description, 
- implemented and tested `moment.locale()`, 
- removed the h2 setting heading, 
- fixed some bug in the previous `RequestUrl()` usage.

Please let me know if there are any further modifications needed.
